src/core/public/application/utils.ts

```diff
@@ -116,20 +124,26 @@ const removeBasePath = (url: string, basePath: IBasePath, origin: string): strin
 };
 
 export function getAppInfo(app: App<unknown> | LegacyApp): PublicAppInfo | PublicLegacyAppInfo {
+  const navLinkStatus =
+    app.navLinkStatus === AppNavLinkStatus.default
+      ? app.status === AppStatus.inaccessible
+        ? AppNavLinkStatus.hidden
+        : AppNavLinkStatus.visible
+      : app.navLinkStatus!;
   if (isLegacyApp(app)) {
```

Comment:
`getAppInfo` now computes the navlink status as it was more or less done in 

https://github.com/elastic/kibana/blob/0247df4cfd689630b174f2f094c35df0a40d16ef/src/core/public/chrome/nav_links/to_nav_link.ts#L38-L41

As `AppServiceStart.applications$` is a public API that is supposed to return the computed state of the apps, I think it make sense, instead of letting the consumers manually apply this logic.